src/main/scala/viper/gobra/reporting/DefaultErrorBackTranslator.scala

```diff
@@ -60,7 +62,13 @@ object DefaultErrorBackTranslator {
     case vprrea.InsufficientPermission(Source(info)) =>
       InsufficientPermissionError(info)
     case vprrea.AssertionFalse(Source(info)) =>
-      AssertionFalseError(info)
+      info.origin match {
+        case origin: AnnotatedOrigin => origin.annotation match {
+          case OverflowCheckAnnotation => OverflowErrorReason(info)
+          case _ => ???
+        }
+        case _ => AssertionFalseError(info)
+      }
```

Comment:
Instead of adding case splits here, could we instead introduce a separate step before in which the annotated errors are translated.

Reply:
Sure, that seems way cleaner!